modin/core/dataframe/pandas/partitioning/axis_partition.py

```diff
@@ -377,6 +377,8 @@ def deploy_splitting_func(
             A list of pandas DataFrames.
         """
         dataframe = pandas.concat(list(partitions), axis=axis, copy=False)
+        # to reduce peak memory consumption
+        del partitions
```

Comment:
Since we manually delete objects now, how does this affect performance? Did you try running any benchmark?

Reply:
> Since we manually delete objects now, how does this affect performance? Did you try running any benchmark?

`del` statement simply removes a reference to the object, allows the garbage collector to free memory earlier, but does not explicitly call it, so there is no direct impact on performance.